### PR TITLE
Adding models and a service folder

### PR DESCRIPTION
**Pull request by macrivello**
The models might not be accurate for our client app, but I'm putting them there as a starting point. 

For example, our API usually returns model fields as IDs, not fully deserialized objects.

### CODE REVIEW

File: webapp/app/models/course.model.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import {Department} from './department.model';
+export class Course {
+  id: number;
+  title: string;
```
Comment (mpjimenez) [inline webapp/app/models/course.model.ts]: missing the number field for our Course model

**Comment by JPautz** [comment]
I agree, let's add the number field for accuracy sake but the implementation may change due to how the API works. This is more of a foundation for members to work off of to stream line initial development.

File: webapp/app/models/course.model.ts (hunk 1 above)
Comment (macrivello) [inline webapp/app/models/course.model.ts]: adding in on new commit

File: webapp/app/models/flowchart.model.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import {FlowchartEntry} from './flowchart-entry.model';
+export class Flowchart {
+  id: number;
+  name: string;
+  user_id: number;
```
Comment (macrivello) [inline webapp/app/models/flowchart.model.ts]: changed from `user: User`

**Pull request merged**